cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java

```diff
@@ -281,7 +323,11 @@ private Table getMetaTable(DatasetContext context) throws IOException, DatasetMa
           ArtifactKey artifactKey = new ArtifactKey(namespace, artifactName);
           Row row = getMetaTable(context).get(artifactKey.getRowKey());
           if (!row.isEmpty()) {
-            addArtifactsToList(artifacts, row);
+            if (order == ArtifactSortOrder.UNORDERED) {
```

Comment:
there's some repeat code here and in `getArtifacts()` above, can we refactor so it's in one method